frontend/src/context/CartContext.test.tsx

```diff
@@ -0,0 +1,250 @@
+import { act, renderHook, type RenderHookResult } from "@testing-library/react";
+import { expect, vi } from "vitest";
+
+import { CartProvider, useCart } from "./CartContext";
+import { getLocalData, setLocalData } from "./utils/localStorage";
+
+vi.mock("./utils/localStorage", () => ({
+  getLocalData: vi.fn(),
+  setLocalData: vi.fn(),
+}));
+
+/**
+ * Helper function to render the CartProvider with a specific initial cart state
+ * @param initialCartValue - The initial cart value to use in the context
+ * @default initialCartValue { nw: {}, pns: {}, wls: {} }
+ */
+function renderComponentWithCart(
+  initialCartValue: Cart | null = { nw: {}, pns: {}, wls: {} },
+): RenderHookResult<ReturnType<typeof useCart>, unknown> {
+  vi.mocked(getLocalData).mockReturnValue(initialCartValue);
+
+  const wrapper = ({ children }: { children: React.ReactNode }) => (
+    <CartProvider>{children}</CartProvider>
+  );
+
+  return renderHook(() => useCart(), { wrapper });
+}
```

Comment:
This file is very large, this could move to a utils file in `/context/utils/renderComponentWithCart.ts`